backend/models/generic.go

```diff
@@ -5,7 +5,11 @@ import (
 )
 
 type Common struct {
-	Id        uint      `json:"id"`
+	Id        uint      `json:"id" gorm:"->"`
```

Comment:
What is this for

Reply:
this just makes it readonly, technically this is covered by line 26 in backend/controller.go but it is just technically a more proper definition of Id as we should never  be manually updating it, so ive left it as so

Reply:
Why aren't we using `gorm:"primaryKey"` again?